scripts/check_licenses.py

```diff
@@ -0,0 +1,92 @@
+"""Check if license fields are valid in all records."""
+
+import asyncio
+import json
+import logging
+import os
+import pathlib
+import time
+
+VALID_LICENSE_IDENTIFIERS = [
+    "CC0-1.0",
+    "GPL-3.0-only",
+    "MIT",
+    "Apache-2.0",
+    "BSD-3-Clause",
+]
+
+logging.basicConfig(level=logging.INFO, format="[%(levelname)s] %(message)s")
+
+
+async def validate_file(path: pathlib.Path) -> int:
+    """Validate a single file."""
+    checks = 0
+    records = await asyncio.get_event_loop().run_in_executor(
+        None, lambda p: json.loads(open(p, "rb").read()), path
+    )
+
+    for record in records:
+        if rec_licenses := record.get("license"):
+            try:
+                attr = rec_licenses["attribution"]
+            except KeyError:
+                recid = record.get("recid", "UNSET")
+                message = f"License field set but without attribution in file {path.name} with recid {recid}!"
+
+                logging.error(message)
+                raise ValueError(message)
+
+            if attr not in VALID_LICENSE_IDENTIFIERS:
+                recid = record.get("recid", "UNSET")
+                message = f"Invalid license `{attr}` identifier in file {path.name} with recid {recid}! "
+
+                logging.error(message)
+                raise ValueError(message)
+            else:
+                checks += 1
+
+    logging.info(f"Successfully validated file {path.name}")
+    return checks
+
+
+async def check_all_paths():
+    """Execute checks on all found files."""
+    start_time = time.perf_counter()
+
+    loop = asyncio.get_event_loop()
+
+    root_path = pathlib.Path(os.getcwd()) / "data" / "records"
+    all_paths = list(root_path.glob("*.json"))
+
+    tasks = [loop.create_task(validate_file(file_path)) for file_path in all_paths]
+    results = await asyncio.gather(*tasks, return_exceptions=True)
+
+    finish_time = time.perf_counter() - start_time
+    logging.info(f"Processed {len(all_paths)} files within {finish_time:.2f} seconds.")
+
+    if any(isinstance(result, Exception) for result in results):
+        errors = len([1 for result in results if isinstance(result, Exception)])
+        logging.error(
+            f"Validation completed with {errors} errors!\n"
+            f"\tPlease ensure the licenses are one of the following: {VALID_LICENSE_IDENTIFIERS}.\n"
+            f"\tIn case it is a valid license, please contact the Open Data Team."
```

Comment:
<!-- TODO's are small, trivial, but necessary changes. Distinguishing todo comments from issues: or suggestions: helps direct the reader's attention to comments requiring more involvement.  -->
**todo:** ‏The last phrase may not be clear to users. Perhaps say: "If you use a valid SPDX license string that is not in the above list, please contact opendata-team@cern.ch".